### PR TITLE
replace deprecated pose2d with pose

### PR DESCRIPTION
**Pull request by ijnek**
Pose2D is deprecated in geometry_msgs, so use Pose.
Alternative would be to have `x`, `y`, `theta` as fields of `BoundingBox2d.msg` and not depend on geometry_msgs/Pose2D.

Resolves: #65

### CODE REVIEW

**Comment by SteveMacenski** [comment]
This seems perfectly reasonable. However, I'd like to hold off until I re-engage with folks on https://github.com/ros2/common_interfaces/issues/179 to see if we can't keep the message around.

**Review by SteveMacenski — APPROVED** [review]

**Comment by ijnek** [comment]
@SteveMacenski updated accordingly.

**Review by SteveMacenski — APPROVED** [review]
LGTM

**Comment by SteveMacenski** [comment]
@Kukanani do you want to take a look

File: msg/Pose2D.msg [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# The 2D position (in pixels) and orientation of the bounding box center (in radians)
```
Comment (Kukanani) [inline msg/Pose2D.msg]:
```suggestion
# Represents a 2D pose (coordinates and a radian rotation). Rotation is positive counterclockwise.
# XY matches the sensor_msgs/Image convention: X is positive right and Y is positive down.
```

**Review by Kukanani — APPROVED** [review]
I love how fast y'all are. One nit to be explicit with documentation, otherwise, looks great.

**Pull request merged**